nix/aws-vm.nix

```diff
@@ -0,0 +1,56 @@
+# this is meant for local dev of the nix flake deployment; it omits ACME and SSL.
+# this variant is automatically selected by Nix when we run the flake with
+#    nixos-rebuild build-vm --flake '.#jl4-aws-2505'
+#    result/bin/ruj-jl4-demo-vm
+#
+# you can also get it to run via libvirt, with virt-install
+
+{ config, pkgs, lib, ... }:
+{
+  virtualisation.vmVariant = {
+    # following configuration is added only when building VM with build-vm
+
+    # we write the IP address of the new instance to /tmp/vm-xchg, so we can ssh into it
+    # we also enable password login for admin.
+    # in the future it would be nice to have the equivalent of virt-install --bridge br0
+
+    virtualisation = {
+      memorySize = 1024; # Use 2048MiB memory.
+      cores = 2;
+      graphics = false;
+      sharedDirectories = {
+        myshare = {
+          source = "/tmp/vm-xchg";
+          target = "/tmp/vm-xchg";
+        };
+      };
+    };
+
+    services.nginx.virtualHosts.${config.networking.domain} = {
+      enableACME = lib.mkForce false;
+      forceSSL = lib.mkForce false;
+    };
+
+    users.groups.admin = {};
+    users.users = {
+      admin = {
+        isNormalUser = true;
+        extraGroups = [ "wheel" ];
+        password = "admin";
+        group = "admin";
+      };
+    };
```

Comment:
out of interested: what is the advantage of this over just using the `root` user?